app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -803,11 +803,7 @@ private boolean processReturnFromFormEntry(int resultCode, Intent intent) {
     private void clearSessionAndExit(AndroidSessionWrapper currentState, boolean shouldWarnUser) {
         currentState.reset();
         if (wasExternal) {
-            if (shouldWarnUser) {
```

Comment:
I am not sure why we are making a distinction here based on `shouldWarnUser`. We end up finishing the activity irrespective of value of `shouldWarnUser` below. So any warning message doesn't show up for external launches anyway.